gluetool/glue.py

```diff
@@ -1978,57 +2066,81 @@ def run_module(self, module_name, module_argv=None, actual_module_name=None, reg
 
         return self.run_modules([step], register=register)
 
-    def module_list(self):
-        # type: () -> List[str]
+    def modules_as_groups(self, modules=None):
+        # type: (Optional[ModuleRegistryType]) -> Dict[str, ModuleRegistryType]
+        """
+        Gathers modules by their groups.
+
+        :rtype: dict(str, dict(str, DiscoveredModule))
+        :returns: dictonary where keys represent module groups, and values are mappings between
+            module names and the corresponding modules.
+        """
+
+        modules = modules or self.modules
+
+        groups = collections.defaultdict(dict)  # type: Dict[str, Dict[str, DiscoveredModule]]
+
+        for name, module_info in modules.iteritems():
+            groups[module_info.group][name] = module_info
+
+        return groups
+
+    def modules_descriptions(self, modules=None, groups=None):
+        # type: (Optional[ModuleRegistryType], Optional[List[str]]) -> str
+        """
+        Returns a string with modules and their descriptions.
+
+        :param dict(str, DiscoveredModule) modules: mapping with modules. If not set, current known modules
```

Comment:
`ModuleRegistryType` ?

Reply:
Will be fixed in a separate patch that should cover use of Mypy types in Sphinx directives in general.